static/js/unread_ops.js

```diff
@@ -58,6 +58,9 @@ export function mark_as_unread_from_here(message_id) {
     const message_ids = message_lists.current.ids_greater_or_equal_than(message_id);
     message_lists.current.prevent_reading();
     message_flags.mark_as_unread(message_ids);
+    if (!message_lists.current.can_mark_messages_read()) {
+        unread_ui.notify_messages_remain_unread();
+    }
```

Comment:
Would it be be more correct to do this in a `success` continuation? In theory, it's possible for the `mark_as_unread` API call to fail, and if it does, I think we don't necessarily want this notice to appear? Right now it's only displayed when there's at least one unread message.

We'd need to pass a callback through to `send_flag_update_for_messages`.

Reply:
That change sounds right to me too :)

Reply:
I was updating this today, and now think it would be better to do this when we get the "update_message_flags" event from the server as that's when we process the unread messages in the UI. Otherwise we potentially have a state where the banner is visible, but the messages haven't been processed as unread yet.

Also, this means that another client in a view where messages aren't marked read (e.g. a narrow with a keyword or sender search) will also make the banner visible when the messages are processed as unread and the view is rerendered.

I'm pushing that change, but I have the success callback version ready to go if there's something I'm overlooking or not thinking about.

Reply:
I agree this is the better way to do this.